framework/src/dgkernels/DGKernel.C

```diff
@@ -202,22 +202,10 @@ DGKernel::computeOffDiagElemNeighJacobian(Moose::DGJacobianType type,unsigned in
 void
 DGKernel::computeOffDiagJacobian(unsigned int jvar)
 {
-  if (jvar == _var.number())
-    computeJacobian();
-  else
-  {
-    // Compute element-element Jacobian
-    computeOffDiagElemNeighJacobian(Moose::ElementElement,jvar);
-
-    // Compute element-neighbor Jacobian
-    computeOffDiagElemNeighJacobian(Moose::ElementNeighbor,jvar);
-
-    // Compute neighbor-element Jacobian
-    computeOffDiagElemNeighJacobian(Moose::NeighborElement,jvar);
-
-    // Compute neighbor-neighbor Jacobian
-    computeOffDiagElemNeighJacobian(Moose::NeighborNeighbor,jvar);
-  }
+  computeOffDiagElemNeighJacobian(Moose::ElementElement,jvar);
+  computeOffDiagElemNeighJacobian(Moose::ElementNeighbor,jvar);
+  computeOffDiagElemNeighJacobian(Moose::NeighborElement,jvar);
+  computeOffDiagElemNeighJacobian(Moose::NeighborNeighbor,jvar);
```

Comment:
It seems like you are now assuming that `DGKernel::computeOffDiagJacobian()` will *never* be called with `jvar==_var.number()`.  This is certainly true based on your change to ComputeFullJacobianThread.C, but I wonder if it might be worth mooseAsserting that jvar != _var.number() in DGKernel::computeOffDiagJacobian() just to be sure someone doesn't call it in the wrong way later?